lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -135,6 +111,83 @@ class QueryBuilder
      */
     private $boundCounter = 0;
 
+    /**
+     * The SELECT parts of the query.
+     *
+     * @var string[]
+     */
+    private $select = [];
+
+    /**
+     * Whether this is a SELECT DISTINCT query.
+     *
+     * @var bool
+     */
+    private $distinct = false;
+
+    /**
+     * The FROM parts of a SELECT query.
+     *
+     * @var From[]
+     */
+    private $from = [];
+
+    /**
+     * The table name for an INSERT, UPDATE or DELETE query.
+     *
+     * @var string|null
+     */
+    private $table;
+
+    /**
+     * The list of joins, indexed by from alias.
+     *
+     * @var array<string, Join[]>
+     */
+    private $join = [];
+
+    /**
+     * The SET parts of an UPDATE query.
+     *
+     * @var string[]
+     */
+    private $set = [];
+
+    /**
+     * The WHERE part of a SELECT, UPDATE or DELETE query.
+     *
+     * @var CompositeExpression|null
+     */
+    private $where = null;
```

Comment:
The `= null` is redundant here.

Reply:
I'm always advocating these default `null` assignments, even if currently redundant, as this will be the correct way to initialize the property to `null` once typed properties will be introduced (failure to do so would throw an `Error` on access), and as such will ease code conversion at that point. But it looks like I'm failing to get converts.

So, half-heartedly done in 3eea67ff8f992f9701fba778373aa1b8b4610849.